src/ThreadedSensitivities.jl

```diff
@@ -1,3 +1,36 @@
 import SciMLBase: ODESolution, build_solution, LinearInterpolation
 using Sundials
 using Base.Threads
+"""
+generate individual sensitivity ODEs for each parameter
+"""
+function generatesensitivityodes(react,sol)
+    sa_list = []
+    y0 = react.y0
+    tspan = react.tspan
+    p = react.p
+    for i in 1:length(p)
+        r = deepcopy(react)
+        jacy!(J::Q2,y::T,p::V,t::Q) where {Q2,T,Q<:Real,V} = jacobiany!(J,y,p,t,r.domain,r.interfaces,nothing)
+        jacp!(J::Q2,y::T,p::V,t::Q) where {Q2,T,Q<:Real,V} = jacobianp!(J,y,p,t,r.domain,r.interfaces,nothing)
+
+        function dsdt(p_index)
+                function dsdt!(ds, s, local_params, t)
+                jy = zeros(length(y0), length(y0))
+                jp = zeros(length(y0), length(p))
+                jacy!(jy, sol(t), p, t)
+                jacp!(jp, sol(t), p, t)
```

Comment:
Don't sure how much it matters but can you save `sol(t)` and avoid calling it multiple times here?

Reply:
Yep